### PR TITLE
#167728017 Implementation of user signup feature

### PR DESCRIPTION
**Pull request by Wokoro**
**What does this PR do?**

  This is an implementation of user signup endpoint, that would allow users of the platform to create new accounts.

  **Description of Task to be completed?**

  Users should be able to register on the platform through a dedicated endpoint, upon successful registration users should receive a message of successful account creation.

  **How should this be manually tested?**

 * Clone repository
 * Install dependencies
 * Navigate to `localhost:3000/api/v1/auth/signup/` using postman
 * Provide require user signup details and send

  **Any background context you want to provide?**

  All dependency should be install.

  **What are the relevant stories?**

 [#167728015](https://www.pivotaltracker.com/story/show/167728015)

  **Screenshots (if appropriate)**
  
signup

  **Questions:**

none

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+end_of_line = lf
+indent_size = 2
+indent_style = space
+insert_final_newline = true
+max_line_length = 80
+trim_trailing_whitespace = true
+
+[{*.json, *.yml}]
+indent_style = space
+indent_size = 2
```
Comment (Abobos) [inline .editorconfig:14]: what's the file for?

Comment (Wokoro) [reply .editorconfig:14]: It is meant to work with our IDE's, so we have a unified coding style and files before pushing to git...

Comment (Wokoro) [reply .editorconfig:14]: On file save, the above settings are affected on the giving file...
Sorry I didn't raise that it was on my mind to....

**Review by Abobos — APPROVED** [review]

File: src/controllers/AuthController.js [hunk 2]
```diff
@@ -13,12 +13,13 @@ export default class AuthController {
    *
    * @param {Function} next - Function to trigger next middleware
    *
-   * @return {Object} Object resoponse with current user created status
+   * @return {Object} Return sucess message and account creation status
    */
   static async signup({ body }, res, next) {
```
Comment (justinefe) [inline src/controllers/AuthController.js]: I think TTL requested that the static method be removed

**Review by justinefe — APPROVED** [review]

**Review by Abobos — APPROVED** [review]

**Review by dieudonneAwa — APPROVED** [review]

File: tests/user.test.js [hunk 3]
```diff
@@ -11,18 +11,19 @@ chai.use(chaiHttp);
...
-        email: '  giftabobo@gmail.com  ',
-        name: ' Bles Abobo',
-        password: 'Blessing9',
+        email: 'wokorosamuel@yahoo.com',
+        name: 'Douye Samuel',
+        password: 'Djkladjkaldfj129',
       })
       .end((err, res) => {
-        expect(res.status).eql(201);
+        expect(res.status).eql(200);
```
Comment (richienabuk) [inline tests/user.test.js]: I think signup should return 201 not 200 because new entry is created.

**Comment by justinefe** [comment]
Cool thanks for the feedback I really don't think sign in is new entry
created like sign up.. Sign in should be OK.. 200

File: tests/user.test.js (hunk 3 above)
Comment (Wokoro) [reply tests/user.test.js]: Thank's for pointing this out, changes would be effected immediately.

**Review by chukwuemekachm — COMMENTED** [review]
I'm curious to know what happens when a user with that email already exists?

Are we hashing a user's password before saving?

**Comment by Wokoro** [comment]
> I'm curious to know what happens when a user with that email already exists?
> 
> Are we hashing a user's password before saving?

Currently, no checks are been done for already existing user account.
The user password is been hashed before saving to the database.

**Review by richienabuk — APPROVED** [review]

**Review by chukwuemekachm — APPROVED** [review]

**Pull request merged**